contracts/Core/Parameters.sol

```diff
@@ -51,71 +51,94 @@ contract Parameters is ACL {
     // keccak256("ASSET_MODIFIER_ROLE")
     bytes32 private constant _ASSET_MODIFIER_HASH = 0xca0fffcc0404933256f3ec63d47233fbb05be25fc0eacc2cfb1a2853993fbbe4;
 
+    //event to be emitted when any governance parameter value changes.
+    event ParameterChanged(address admin, string parameterName, uint256 valueChangedFrom, uint256 valueChangedTo, uint256 timestamp);
+
+    //event to be emitted when escape hatch is disabled
+    event EscapeHatchDisabled(address admin, string message, bool status, uint256 timestamp);
+
     function setPenaltyNotRevealNum(uint256 _penaltyNotRevealNumerator) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        emit ParameterChanged(msg.sender, "Penalty Not Reveal Numerator", penaltyNotRevealNum, _penaltyNotRevealNumerator, block.timestamp);
```

Comment:
Set the string as the parameter name "penaltyNotRevealNum" instead of "Penalty Not Reveal Numerator"